prototypes/marimekko-chart/src/main.js

```diff
@@ -0,0 +1,685 @@
+                        reverseBars, 
+                        reverseSeg, 
+                        sortBar,
+                        sortSeg, 
+                        size)
+    {
+        if (state.preventRender) {
+            // Early return if the state currently disallows rendering.
+            return;
+        }
+
+        const onSelection = ({ dragSelectActive }) => {
+            state.preventRender = dragSelectActive;
+        };
+
+        let modProperty = {
+            xAxisMode: xAxisMode, 
+            chartMode: chartMode,
+            titleMode: titleMode, 
+            labelMode: labelMode,
+            labelBars: labelBars,
+            labelSeg: labelSeg,
+            numeric: numeric,
+            percentage: percentage,
+            category: category,
+            reverseBars: reverseBars,
+            reverseSeg: reverseSeg,
+            sortBar: sortBar,
+            sortSeg: sortSeg
+        };
+        
+        /**
+         * Check for any errors.
+         */
+        let errors = await dataView.getErrors();
+        if (errors.length > 0) {
+            mod.controls.errorOverlay.show(errors, "dataView");
+            // TODO clear DOM
+            return;
+        }
+
+        mod.controls.errorOverlay.hide("dataView");
+
+        // Return and wait for next call to render when reading data was aborted.
+        // Last rendered data view is still valid from a users perspective since
+        // a document modification was made during a progress indication.
+        // Hard abort if row count exceeds an arbitrary selected limit
+        const xLimit = 1000;
+        const colorLimit = 100;
+        const colorCount = (await dataView.hierarchy("Color")).leafCount;
+        const xCount = (await dataView.hierarchy("Categories")).leafCount;
+        if (colorCount > colorLimit || xCount > xLimit) {
+            canvasDiv.remove();
+                        mod.controls.errorOverlay.show("The resulting data view exceeded the size limit.", "dataViewSize1");
+            if (xCount > xLimit) {
+                mod.controls.errorOverlay.show(
+                    `Maximum number of categories is limited to ${xLimit}. Try aggregating the expression further.`,
+                    "dataViewSize2"
+                );
+            } else {
+                mod.controls.errorOverlay.hide("dataViewSize2");
+            }
+                        if (colorCount > colorLimit) {
+                mod.controls.errorOverlay.show(
+                    `Maximum number of colors is limited to ${colorLimit} for readability reasons.`,
+                    "dataViewSize3"
+                );
+            } else {
+                mod.controls.errorOverlay.hide("dataViewSize3");
+            }
+                        return;
+        } else {
+            mod.controls.errorOverlay.hide("dataViewSize1");
+            mod.controls.errorOverlay.hide("dataViewSize2");
+            mod.controls.errorOverlay.hide("dataViewSize3");
+        }
```

Comment:
The first argument in the show method can be an array of strings as well. Utilizing that would simplify this part of the code.
https://tibcosoftware.github.io/spotfire-mods/api-docs/interfaces/ErrorOverlay.html#show